pkg/helm/controller/reconcile.go

```diff
@@ -343,12 +343,16 @@ func hasHelmUpgradeForceAnnotation(o *unstructured.Unstructured) bool {
 }
 
 func (r HelmOperatorReconciler) updateResource(o runtime.Object) error {
-	return r.Client.Update(context.TODO(), o)
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		return r.Client.Update(context.TODO(), o)
+	})
```

Comment:
Now that we're no longer using the `GenerationChangedPredicate`, we get more back-to-back reconciliations, so we need to retry on conflict since the cache may not have updated in time to return the latest version of the CR.